objects/src/accounts/builder/mod.rs

```diff
@@ -158,6 +215,36 @@ impl AccountBuilder {
     pub fn build(self) -> Result<(Account, Word), AccountError> {
         let (init_seed, vault, code, storage) = self.build_inner()?;
 
+        // Anchor block hash and anchor epoch must only be set when building a new account.
+        if self.anchor_block_hash == Digest::default() {
+            return Err(AccountError::BuildError(
+                "anchor block hash must be set to a `Digest` different from the empty value".into(),
+                None,
+            ));
+        }
```

Comment:
Good point! 

I think with regards to the previous comment about `anchor_block_hash` and `anchor_block_number`, we might need those to construct a genesis account, although we can also create some special `BlockHeader` - many options here probably. We could add a field in the builder to disable this `anchor_block_hash != EMPTY_WORD` check and then we should be able to build genesis accounts with the builder as well (and just use anchor_epoch = 0 for them).

Otherwise we can indeed build them in a more manual way as well, but since `Account::initialize_from_components` is now private, we would have to do genesis building either in `miden-objects` or use the `AccountBuilder` with the above changes from wherever (e.g. miden-node).